### DISCUSSION TITLE
Organização das atividades

### DISCUSSION THREAD

**Discussion by henriqgoncalvs** [diego3g/rsxp-2023#17] [General]
## Como vocês acham que é a melhor de forma de organizar as tasks de FE e BE?

De cara eu já penso em usar o [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/quickstart-for-projects) aqui dentro do Github mesmo usando issues, daí quem for trabalhar na task avisa que tem interesse e que ta trabalhando naquilo (como funciona na maioria dos projetos open source aqui no git). Aí pode criar um project pra FE e outro pra BE, por exemplo, não sei tô jogando ideias.

Mas a gente pode pensar em outra solução tipo Trello ou Linear, por exemplo.

---

Um exemplo legal que eu já vi funcionando com essa organização usando issue + PR foi o ariakit. Segue o link pra vcs verem também

https://github.com/ariakit/ariakit/issues/939

**Comment by diego3g** [comment]
Essa é uma dúvida minha também, por exemplo:

- O que acontece se duas pessoas começarem a desenvolver a mesma feature?
- O que acontece se uma pessoa marcar que quer desenvolver uma feature, mas sumir e não comentar mais sobre?

Tenho meus pontos, mas acho que um projeto aqui no Github vai ser o melhor formato sim! 

Pretendo criar uma v1 da estrutura do app e criar uma lista de tasks para trabalharmos juntos.

**Reply by LuizHAP** [reply]
@diego3g acho que com a lista de tarefas podemos criar por pequenos cards e utilizar o próprio Kanban do Github. Dividir em tasks pequenas e independentes

**Reply by henriqgoncalvs** [reply]
@diego3g Acho que esse primeiro ponto é mais fácil de gerenciar, a primeira pessoa que se voluntariar pra pegar a task, tem prioridade nela.

Já o segundo ponto, o que a gente pode fazer é definir algumas pessoas que ficam responsáveis por esse gerenciamento, aí fica como responsabilidade ir checando ali na issue o andamento de tempos em tempos, até manter contato direto pelo discord. Dá pra pensar em estratégias

**Reply by d3vlopes** [reply]
Sobre a questão de duas pessoas fazer a mesma feature o Github Projects já resolve isso, onde quando alguém começar a trabalhar em uma issue ela pode alterar o status para "in progress", isso mostra para outra pessoa que alguém já esta trabalhando naquela feature.

Sobre o outro ponto, acho importante definir algumas questões antes de começar o projeto, como definir que todo dia o código produzido naquele dia deve ser enviado em uma branch para o repositório original, assim tem como verificar as branchs ativas e se alguém sumir outra pessoa pode continuar de onde foi parado.

**Reply by JonasJs** [reply]
**O que acontece se uma pessoa marcar que quer desenvolver uma feature, mas sumir e não comentar mais sobre?**
Será que não podemos criar um esquema de "auto reporte", como uma Daily da demanda ? a cada dia, adicionamos em que etapa esta ou até mesmo se esta com algum problema. (podemos ter uma label)
Assim podemos a cada x dias, fazer uma varredura nos card e analisar o historico, caso esteja muito tempo sem o report, podemos marcar como abandonada.

**Reply by henriqgoncalvs** [reply]
Gostei das duas sugestões, acho que tira um pouco da necessidade de ter alguém constantemente olhando todos as issues. O que fico pensando ainda é que de toda forma vai precisar ter pessoas encarregadas de ficar fazendo essas verificações, principalmente na ideia do @JonasJs 

Acho que uma coisa que pode ajudar muito é criarmos um server no discord, ou um canal no server da RS pra ir falando. Se alguém tiver um obstáculo, ou precisar abandonar a task que tava mexendo por algum motivo, desse jeito acho que a gente consegue mitigar a chance de ter pessoas que vão simplesmente desaparecer depois de pegarem uma task.

**Reply by JonasJs** [reply]
@hnqg acho que essa função, pode ser colaborativa, caso alguém pegue uma demanda abandonada, colocamos como uma boa prática validar os critérios para dar uma demanda como abandonada. Caso alguém tenha movido errado, o novo contribuidor que possa pegar essa demanda, pode corrigir o erro, fazendo essa verificação.
O mesmo vale para alguém que estava fazendo a demanda e viu que sua demanda foi puxada de forma indevida.

Precisamos apenas criar um documento de boas praticas e recomendações.

**Edit**: Publiquei a ideia do documento para que possamos evoluir para outros tópicos, pois acredito que serviria como direção para qualquer duvida de processos que os contribuidores poderiam ter.
[Documentação de boas práticas e recomendações gerais. #18](https://github.com/diego3g/rsxp-2023/discussions/18)

**Comment by nitoba** [comment]
Acho que uma boa ideia até pra fazer um pouco de network e talvez desenvolver mais rápido as features é também desenvolver as tarefas fazendo pair programming. Assim mais pessoas que queriam trabalhar numa mesma feature podem ter essa oportunidade em conjunto com outras. O que acham?

**Reply by henriqgoncalvs** [reply]
Creio que funciona sim mano, não sei se é uma coisa que a gente vai ter muito controle, mas pode ser algo que incentivamos nesse projeto. Se 2 ou mais pessoas tiverem interesse na mesma task, elas sinalizam na issue e depois ficam responsáveis por correr atrás de se juntarem pra fazer o pair.

Se for pra frente a ideia do discord, acho que já facilita muito isso

**Reply by d3vlopes** [reply]
O problema que vejo em fazer pair programming é que demanda muita disponibilidade das duas partes, além de precisar ter um planejamento de horário.

**Reply by nitoba** [reply]
Acho que quando duas pessoas sinalizarem que querem trabalhar na mesma task, o pair programming pode ser acordado entre as partes. Ai fica a cargo delas decidirem essa questão de horário ou irem pegando commits uma das outras. Não necessariamente precisam está sempre codando juntas

**Reply by rafael-hc** [reply]
Acho esse questão do pair programming super interessante, pois abre possibilidades de pessoas com menos experiências poderem aprender e contribuir. Eu por exemplo, não tenho experiência de trabalhar com times, estou estudando a um bom tempo, mas ainda não tive essa experiência.
Tenho muita vontade de experimentar esse ambiente e aprender. Mesmo com um certo receio de não dar conta de pegar uma task e finalizar, eu quero pegar, ir para cima e me desafiar.

**Comment by diego3g** [comment]
Galera criei um board simples de tarefas por enquanto via issues pra irmos nos organizando: #49 

Vou fechar essa discussão por enquanto pra termos mais espaço pra novas discussões 🤗
